pkg/vault/vault.go

```diff
@@ -56,12 +56,37 @@ func NewVault(cfg Config) (*Vault, error) {
 	return vault, nil
 }
 
+func validateVaultConfig(cfg Config) error {
```

Comment:
nit: Typically we do validation of config via `(*cfg).Validate` method, before passing config to `NewVault`. In my opinion doing it in `NewVault` makes more sense, but it's better to be consistent with the rest of Mimir's codebase and follow the same pattern.

This is the main entry point for validation: https://github.com/grafana/mimir/blob/9e8e46a0f59643ae432f8e789be8d82a2ce330df/pkg/mimir/mimir.go#L211-L210

One detail: if you do this, then let's not validate URL/Token/MountPath if Vault is disabled.